libs/eventually-pg/src/PostgresMessageQueue.ts

```diff
@@ -84,34 +81,22 @@ export const PostgresMessageQueue = <M extends Messages>(
         if (next.length === 0) {
           log().yellow().trace(`No messages available for stream: ${stream}`);
           await client.query("ROLLBACK");
-          return;
+          return false;
         }
 
-        // update lock to prevent other consumers from accessing it
-        const message = next[0];
-        await client.query(
-          `
-          UPDATE "${table}" SET locked_until = NOW() + INTERVAL '1 millisecond' * $1
```

Comment:
trying to find out where is the lock/timeout set before processing

Reply:
before the changes I made, that update would get committed if the handler failed.  I vote that we do not implement this timeout and just rollback then return the error to the caller so the caller can decide to retry immediately, retry with delay, circuit break, etc.

Reply:
ok... this works but you need a broker to coordinate retries or put that responsibility on the consumers... part of this exercise is to determine what works best, push or pull model.

Reply:
we probably don't need the timeout param in the interface anymore